tests/assets/test_digital_assets.py

```diff
@@ -0,0 +1,163 @@
+import pytest
+from ..db.conftest import inputs
+
+
+@pytest.mark.usefixtures('inputs')
+def test_asset_transfer(b, user_vk, user_sk):
+    from bigchaindb.models import Transaction
+
+    tx_input = b.get_owned_ids(user_vk).pop()
+    tx_create = b.get_transaction(tx_input.txid)
+    
+    tx_transfer = Transaction.transfer(tx_create.to_inputs(), [user_vk], tx_create.asset)
+    tx_transfer_signed = tx_transfer.sign([user_sk])
+
+    assert tx_transfer_signed.validate(b) == tx_transfer_signed
+    assert tx_transfer_signed.asset.data_id == tx_create.asset.data_id
+
+
+def test_validate_bad_asset_creation(b, user_vk):
+    from bigchaindb_common.exceptions import AmountError
+    from bigchaindb.models import Transaction
+
+    # `divisible` needs to be a boolean
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.divisible = 1
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        tx_signed.validate(b)
+
+    # `refillable` needs to be a boolean
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.refillable = 1
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # `updatable` needs to be a boolean
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.updatable = 1
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # `data` needs to be a dictionary
+    tx = Transaction.create([b.me], [user_vk])
+    tx.asset.data = 'a'
+    tx_signed = tx.sign([b.me_private])
+    with pytest.raises(TypeError):
+        b.validate_transaction(tx_signed)
+
+    # TODO: Check where to test for the amount
```

Comment:
Lets leave it here and figure it out when implementing divisible assets.
But I guess the only two places where you can check that is on `Transaction.validate` or `Transaction.__init__` because those are the only places where you have all the information needed to validate the amounts